app/scripts/controllers/metametrics-data-deletion/metametrics-data-deletion.ts

```diff
@@ -0,0 +1,138 @@
+import { ObservableStore } from '@metamask/obs-store';
+import MetaMetricsController from '../metametrics';
+import {
+  createDataDeletionRegulationTask,
+  fetchDeletionRegulationStatus,
+} from './services/services';
+
+export type DataDeleteDate = string | undefined;
+export type DataDeleteRegulationId = string | undefined;
+
+export type RegulationId = Record<string, string>;
+export type CurrentRegulationStatus = Record<string, Record<string, string>>;
+export type DeleteRegulationAPIResponse = {
+  data: Record<string, RegulationId | CurrentRegulationStatus>;
+};
+
+export enum responseStatus {
+  ok = 'ok',
+  error = 'error',
+}
+
+export type DataDeletionResponse = {
+  status: responseStatus;
+  metaMetricsDataDeletionStatus?: string | null;
+  error?: string;
+};
+
+export type MetaMetricsDataDeletionState = {
+  metaMetricsDataDeletionId: DataDeleteRegulationId;
+  metaMetricsDataDeletionDate: DataDeleteDate;
+};
+
+const defaultState: MetaMetricsDataDeletionState = {
+  metaMetricsDataDeletionId: undefined,
+  metaMetricsDataDeletionDate: undefined,
+};
+
+/**
+ * Controller responsible for maintaining
+ * state related to Metametrics data deletion
+ */
+export default class MetaMetricsDataDeletionController {
+  /**
+   * Observable store containing controller data.
+   */
+  store: ObservableStore<MetaMetricsDataDeletionState>;
+
+  private metaMetricsController: MetaMetricsController;
+
+  /**
+   * Constructs a DeleteMetaMetricsData  controller.
+   *
+   * @param options - the controller options
+   * @param options.initState - Initial controller state.
+   * @param options.metaMetricsController
+   */
+  constructor({
+    initState,
+    metaMetricsController,
+  }: {
+    initState: MetaMetricsDataDeletionState;
+    metaMetricsController: MetaMetricsController;
+  }) {
+    this.store = new ObservableStore({
+      ...defaultState,
+      ...initState,
+    });
+    this.metaMetricsController = metaMetricsController;
+  }
+
+  _formatDeletionDate() {
+    const currentDate = new Date();
+    const day = currentDate.getUTCDate();
+    const month = currentDate.getUTCMonth() + 1;
+    const year = currentDate.getUTCFullYear();
+
+    // format the date in the format DD/MM/YYYY
+    return `${day}/${month}/${year}`;
+  }
+
+  async createMetaMetricsDataDeletionTask(): Promise<DataDeletionResponse> {
+    const segmentSourceId = process.env.SEGMENT_DELETE_API_SOURCE_ID;
```

Comment:
The definition of these constant variables can happen outside of this method and outside of the class. We don't need to re-define them every time the method is called.